### PR TITLE
Merge all the stacked commits to develop

### PR DESCRIPTION
**Pull request by Maghwyn**
Regroup all the commits from each develop-GX.

Due to a large number of commits, it's becoming quite difficult to develop on our own branch with conflicts appearing every now and then because the auto-develop is being merged with commits from other groups due to those conflicts. This PR has the sole goal of being merged as soon as possible to unlock everything.

Features :

- Rework of the pixi-lib from v1 to v2.
- Advanced QRCode implementation.
- Materials list and permission fix.
- Advanced blog features.
- User profile edit and display.
- Course list

Warning :

This PR might break some css, one of the flex tailwind property was manually changed and caused some issue.

### CODE REVIEW

File: back-end/src/base/materials/materials.module.ts [hunk 1]
```diff
@@ -7,9 +7,18 @@ import { UsersRepository } from 'src/base/users/users.repository';
 import { MaterialsRepository } from 'src/base/materials/materials.repository';
 import { MaterialsService } from 'src/base/materials/materials.service';
 import { MaterialsController } from 'src/base/materials/materials.controller';
+import { JwtModule } from '@nestjs/jwt';
+import { config } from '@/config/config';
 
 @Module({
-	imports: [DatabaseModule, forwardRef(() => AuthModule)],
+	imports: [
```
Comment (Maghwyn) [inline back-end/src/base/materials/materials.module.ts]: Unless you're using a jwt with the materiels, this shouldn't be there.

File: back-end/src/base/materials/materials.repository.ts [hunk 2]
```diff
@@ -15,14 +15,17 @@ export class MaterialsRepository {
 		return this.materials.find().toArray();
 	}
 	async createMaterial(query: Material) {
-		return this.materials.insertOne(query);
+		this.materials.insertOne(query);
+		//Return the new material
+		return this.materials.findOne(query);
```
Comment (Maghwyn) [inline back-end/src/base/materials/materials.repository.ts:20]: That's a hack of the repository purpose, but it's not causing any bug.

File: back-end/src/base/sprints/sprints.repository.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+import { Filter, UpdateFilter, FindOneAndUpdateOptions, Db } from 'mongodb';
+
+import { Sprint } from 'src/base/sprints/interfaces/sprints.interface';
+
+@Injectable()
+export class SprintsRepository {
+	constructor(@Inject('DATABASE_CONNECTION') private db: Db) {}
+
+	get groups() {
+		return this.db.collection<Sprint>('groups');
```
Comment (Maghwyn) [inline back-end/src/base/sprints/sprints.repository.ts:11]: Why is it called groups if it's sprints ?

File: front-end/src/components/NewGroup.vue [hunk 4]
```diff
@@ -0,0 +1,72 @@
+<template>
+    <div class="popupcontainer">
+        <div class="fond"></div>
+        <form class="newGroupForm">
+            <button class="iconBig" type="button" @click="$emit('close')"><img src="https://cdn-icons-png.flaticon.com/512/2976/2976286.png" alt="Close"></button>
```
Comment (Maghwyn) [inline front-end/src/components/NewGroup.vue:5]: Careful about the link, make sure it doesn't expire.

File: front-end/src/components/NewSprint.vue [hunk 5]
```diff
@@ -0,0 +1,84 @@
+<template>
+    <div class="popupcontainer">
+        <div class="fond"></div>
+        <form class="newSprintForm">
+            <button class="iconBig" type="button" @click="$emit('close')"><img src="https://cdn-icons-png.flaticon.com/512/2976/2976286.png" alt="Close"></button>
```
Comment (Maghwyn) [inline front-end/src/components/NewSprint.vue:5]: Same, check if the link may be subject to expiration.

File: front-end/src/components/RollCallComponents/CallsComponent.vue [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+	</div>
+</template>
+
+
+<script lang="ts">
+import Qrcode from 'vue-qrcode';
+import { http } from '@/api/network/axios';
+import { withErrorHandler } from '@/utils/storeHandler';
+
+let url = '';
```
Comment (Maghwyn) [inline front-end/src/components/RollCallComponents/CallsComponent.vue:19]: Why are these defined both in the data and outside the component as global var ?

File: front-end/src/components/RollCallComponents/GroupsComponent.vue [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+			Empty Groups
+		</button>
+	</div>
+</template>
+
+<script lang="ts">
+import { http } from '@/api/network/axios';
+import { withErrorHandler } from '@/utils/storeHandler';
+
+let courseId = '';
```
Comment (Maghwyn) [inline front-end/src/components/RollCallComponents/GroupsComponent.vue:53]: Same, i'm not i understand the necessity of having them both inside and outside.
Also, those outside are definitely not reactive.

File: front-end/src/components/RollCallComponents/MakeGroupsComponent.vue [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+			<button class="bg-blue-500 text-white py-2 px-4 rounded mt-4">Former les groupes</button>
+		</a>
+	</div>
+</template>
+
+<script lang="ts">
+import { http } from '@/api/network/axios';
+import { withErrorHandler } from '@/utils/storeHandler';
+
+let groupUrl = '';
```
Comment (Maghwyn) [inline front-end/src/components/RollCallComponents/MakeGroupsComponent.vue:13]: Same as the previous message about scoping.

File: front-end/src/components/RollCallComponents/StudentListComponent.vue [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+			</tbody>
+		</table>
+	</div>
+</template>
+
+<script lang="ts">
+import { http } from '@/api/network/axios';
+import { withErrorHandler } from '@/utils/storeHandler';
+
+let courseId = '';
```
Comment (Maghwyn) [inline front-end/src/components/RollCallComponents/StudentListComponent.vue:47]: And another one.

File: front-end/src/store/interfaces/auth.interfaces.ts [hunk 10]
```diff
@@ -29,8 +29,16 @@ export interface UserCreds {
...
+}
+
 export interface User {
-	role?: string;
+	id?: string;
+	_id?: string;
+	role?: Roles;
 	profile?: UserProfile;
 	schoolProfile?: UserSchoolProfile;
 	businessProfile?: UserBusinessProfile;
```
Comment (Maghwyn) [inline front-end/src/store/interfaces/auth.interfaces.ts:46]: Need an explanation about the myArticles and events, they shouldn't be on the user itself.

File: front-end/src/store/modules/material.store.ts [hunk 11]
```diff
@@ -0,0 +1,90 @@
...
+import { Material, MaterialStore } from '@/store/interfaces/material.interface';
+
+export const useMaterialStore = defineStore('materialStore', {
+	state: (): {
+		filter: { input: string; site: string; type: string; state: string };
+		input: string;
+		materials: Array<Material>;
+	} => {
+		return {
+			materials: [],
```
Comment (Maghwyn) [inline front-end/src/store/modules/material.store.ts:16]: That's fine, but then you can't reset your store on logout.
Just letting you known.

File: front-end/src/store/modules/agility.store.ts [hunk 12]
```diff
@@ -1,17 +1,74 @@
...
+	apiTrySaveProjectMeta,
+	apiTryGetRoomAccess,
+} from '@/api/agility-req';
+import { withErrorHandler } from '@/utils/storeHandler';
 
 export const useAgilityStore = defineStore('agility', {
 	state: (): AgilityStore => {
 		return {
-			metaProjects: [],
+			projects: [],
```
Comment (Maghwyn) [inline front-end/src/store/modules/agility.store.ts]: Can't reset the store.

File: front-end/src/store/modules/project.store.ts [hunk 13]
```diff
@@ -1,194 +1,137 @@
...
-				createdAt: new Date().toLocaleDateString(),
-				updatedAt: new Date().toLocaleDateString(),
-			},
-			viewportBounds: {
-				posX: undefined,
-				posY: undefined,
-				scaleX: undefined,
-				scaleY: undefined,
-			},
+			scene: null,
```
Comment (Maghwyn) [inline front-end/src/store/modules/project.store.ts]: Can't reset the store

**Review by Maghwyn — COMMENTED** [review]
up

**Review by Kevw94 — APPROVED** [review]

**Review by Iagomorph — APPROVED** [review]

**Review by Massimiliano-HA — APPROVED** [review]

**Pull request merged**